src/transformers/trainer.py

```diff
@@ -653,7 +704,13 @@ def _prediction_loop(
                     else:
                         label_ids = np.append(label_ids, inputs["labels"].detach().cpu().numpy(), axis=0)
 
-        if is_tpu_available():
+        if self.args.local_rank != -1:
```

Comment:
Combination of this and line 464 means that this part is only called in the rank 0 process (self.args.local_rank==0). This will cause evaluation in the main training loop to only be run on a subset of the evaluation dataset. Need to change main training loop in order to accommodate this.

Reply:
In addition, lines 664-667 should be updated to move the model to DDP as well (if it's not already distributed), if I'm not mistaken. This already happens in the `train()`, but it won't have happened if you use `predict()` or `evaluate()` directly.

Reply:
Yes that is correct, the PR is still incomplete